### PR TITLE
Feat/create badge component

### PR DESCRIPTION
**Pull request by arwehrman**

### CODE REVIEW

File: src/packages/solid/components/Badge/Badge.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  iconHeight: number;
+  /**
+   * Object containing all properties supported in the [Icon component](?path=/docs/components-icon--icon)
+   */
+  icon: Partial<IconProps>;
+};
+
+const Badge: Component<BadgeProps> = (props: BadgeProps) => {
+  // not implemented, thinking about which option would be best
+  const iconAlignStyles = (props) => {
```
Comment (arwehrman) [inline src/packages/solid/components/Badge/Badge.tsx:53]: started to add this but wanted to wait to see which way we wanted to go before working out all the details for this work

File: src/packages/solid/components/Badge/Badge.styles.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import theme from 'theme';
+import { getHexColor } from '../../../../shared/utils/index'; // TODO ts path aliasing
+
+// styles in LUI for Badge
```
Comment (arwehrman) [inline src/packages/solid/components/Badge/Badge.styles.ts:21]: just notes, will remove before merging pr

File: src/packages/solid/components/Badge/Badge.tsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  // NOTE: directive needs to be before props
+  return (
+    <node
+      use:withPadding={styles.padding}
+      {...props}
+      color={styles.color}
+      borderRadius={styles.borderRadius}
+      border={styles.border}
+      width={styles.width}
+      height={styles.height}
```
[the remark this answers is not in the document]
Comment (chiefcll) [reply src/packages/solid/components/Badge/Badge.tsx:74]: And on further thought, now that we have two elements don't use withPadding at all. We'll need another container element and flex now supports margin so we should be good.

File: src/packages/solid/components/Badge/Badge.tsx (hunk 1 above)
Comment (chiefcll) [reply src/packages/solid/components/Badge/Badge.tsx:53]: I would go with the two <Show> statements and then change node to View and flex container - Solid needs an updateLayout solution for this now that you have two kids we need to figure out the total width

**Comment by chiefcll** [comment]
Can we merge now?

**Pull request merged**